pinot-integration-test-base/src/test/java/org/apache/pinot/integration/tests/BaseClusterIntegrationTest.java

```diff
@@ -724,6 +749,30 @@ protected long getCurrentCountStarResult(String tableName) {
     return 0;
   }
 
+  protected void waitForMinionTaskCompletion(String taskId) {
+    TestUtils.waitForCondition(aVoid ->
+            _controllerStarter.getHelixTaskResourceManager().getTaskState(taskId) == TaskState.COMPLETED,
+        300_000L, "Failed to complete the task " + taskId);
```

Comment:
nit: it might be useful to allow configuring this timeout via a method parameter?